ui/webui/src/components/localization/InstallationLanguage.jsx

```diff
@@ -236,7 +236,19 @@ class LanguageSelector extends React.Component {
                                 .catch(this.props.onAddErrorNotification);
                         this.setState({ lang: item });
                         this.updateNativeName(localeItem);
-                        window.location.reload(true);
+                        fetch("po.js").then(response => response.text())
+                                .then(body => {
+                                    // always reset old translations
+                                    cockpit.locale(null);
+                                    // en_US is always null
+                                    if (body.trim() === "") {
+                                        cockpit.locale(null);
+                                    } else {
+                                        // eslint-disable-next-line no-eval
+                                        eval(body);
```

Comment:
I wonder - can the contents of this be influenced via say crafted translation strings ? Eq. someone injecting JS instead of a translation in Weblate & it then being executed via this eval ?

Reply:
We have checks for something like that in Weblate, but it's unclear how much that would protect us.